hoot-core-test/src/test/cpp/hoot/core/algorithms/changeset/ChangesetDeriverTest.cpp

```diff
@@ -53,17 +52,12 @@ class ChangesetDeriverTest : public HootTestFixture
   QMap<Change::ChangeType, QList<long>> _getChangeset(
     const QString& input1, const QString& input2, const bool allowRefDelete = true)
   {
-    CalculateHashVisitor2 hashVis;
-
     OsmMapPtr map1(new OsmMap());
     // each dataset needs to have a different input status
     OsmMapReaderFactory::read(map1, input1, true, Status::Unknown1);
-    // nodes have to have hash values for sorting
-    map1->visitRw(hashVis);
 
     OsmMapPtr map2(new OsmMap());
     OsmMapReaderFactory::read(map2, input2, true, Status::Unknown2);
-    map2->visitRw(hashVis);
 
```

Comment:
I made `ElementComparer` auto add the node hash if its missing, so no longer need to add it beforehand. There are similar changes to this throughout.